File: src/backend/_check_fk.py
from subprocess import run

r = run(['node', '-e', """
const { Pool } = require('pg');
const pool = new Pool({ host: 'localhost', port: 5432, database: 'cabinet_factory', user: 'postgres', password: 'postgres' });
pool.query("SELECT tc.table_name, ccu.table_name AS foreign_table_name FROM information_schema.table_constraints AS tc JOIN information_schema.constraint_column_usage AS ccu ON ccu.constraint_name = tc.constraint_name WHERE tc.constraint_type = 'FOREIGN KEY' AND tc.table_schema = 'public' ORDER BY tc.table_name", [], function(err, res) { if (err) console.error(err.message); else console.log(JSON.stringify(res.rows)); pool.end(); });
"""], capture_output=True, timeout=15, shell=True)

print(r.stdout.decode('utf-8', errors='replace'))
print(r.stderr.decode('utf-8', errors='replace')[:200])

File: src/backend/_check_fk.js
const { Pool } = require('pg');
const pool = new Pool({ host: 'localhost', port: 5432, database: 'cabinet_factory', user: 'postgres', password: 'postgres' });
pool.query("SELECT tc.table_name, ccu.table_name AS foreign_table_name FROM information_schema.table_constraints AS tc JOIN information_schema.constraint_column_usage AS ccu ON ccu.constraint_name = tc.constraint_name WHERE tc.constraint_type = 'FOREIGN KEY' AND tc.table_schema = 'public' ORDER BY tc.table_name", [], function(err, res) {
    if (err) { console.error(err.message); pool.end(); return; }
    const rows = res.rows;
    const grouped = {};
    for (const r of rows) {
        if (!grouped[r.table_name]) grouped[r.table_name] = [];
        grouped[r.table_name].push(r.foreign_table_name);
    }
    for (const [tbl, parents] of Object.entries(grouped)) {
        console.log(tbl + ' -> ' + parents.join(', '));
    }
    pool.end();
});
